boards.txt

```diff
@@ -3522,4 +3522,72 @@ sparkfun_lora_gateway_1-channel.menu.UploadSpeed.512000.upload.speed=512000
 
 ##############################################################
 
+ttgo-t-watch.name=TTGO T-Watch
+
+ttgo-t-watch.upload.tool=esptool_py
+ttgo-t-watch.upload.maximum_size=6553600
+ttgo-t-watch.upload.maximum_data_size=4521984
+ttgo-t-watch.upload.wait_for_upload_port=true
+
+ttgo-t-watch.serial.disableDTR=true
+ttgo-t-watch.serial.disableRTS=true
+
+ttgo-t-watch.build.mcu=esp32
+ttgo-t-watch.build.core=esp32
+ttgo-t-watch.build.variant=t-watch
```

Comment:
do not use `-` in the names of the variant and board. those things are later turned into defines and can cause issues.